### DISCUSSION TITLE
How to Return cookie value from cookie_jar file as string

### DISCUSSION THREAD

**Discussion by MohdRashid01** [cypress-io/cypress#18532] [Questions and Help] [answered]
Hi All

I have one file that name is cookie_jar file in that file there is a cookie value stored in it that i need return in it as string way i have written one command but dont know how to put in as a bash script or shell script. So that my script should return as a token values. Here is the code

cookie_value = "'cookie: PF='$PF'; p_lplcwmsk='$p_lplcwmsk'; BIGipServerNP_QA_QFLOGINQAS0_7011='$BIGipServerNP_QA_QFLOGINQAS0_7011'; aatoken='$aatoken'; UserName='$UserName'; Auth='$Auth'; XSRF-TOKEN='$XSRF-TOKEN'; z='$z'; BN20='$BN20'; UserFirmId='$UserFirmId'; TS01a1bccb='$TS01a1bccb'; TS01e14722='$TS01e14722';"

echo cookie_value

please tell me how to write in script so that it can return string from cookie jar file as cookie value i want it that only from script

**Comment by jennifer-shehane** [comment, marked as answer]
Hey, this looks like a bash/curl cookie-jar question rather than a Cypress one, so a general forum like Stack Overflow would get you a faster answer. A couple of quick fixes either way:

- In bash, no spaces around =, and quote the var: cookie_value="..." then echo "$cookie_value".
- To build the string from a curl cookie jar (tab-separated domain flag path secure expiry name value):
- 
```js
cookie_value=$(awk -F'\t' '!/^#/ && NF>=7 {printf "%s%s=%s", sep, $6, $7; sep="; "}' cookie_jar)
echo "Cookie: $cookie_value"
```

If you actually meant to do this inside a Cypress test, use cy.getCookies():

```js
cy.getCookies().then((cookies) => {
  const cookieHeader = cookies.map((c) => `${c.name}=${c.value}`).join('; ')
})
```
